File: Automation/001-AutomationFramework/C#/AutomationFramework/Common/CommonCore/TestInMemoryParameters.cs
using Newtonsoft.Json.Linq;
using System.Xml;
using Utilities;
using static NUnit.Framework.TestContext;

namespace CommonCore.Tests
{
    public sealed class TestInMemoryParameters
    {
        #region Fields and Properties
        private static TestInMemoryParameters instance = null;
        private static readonly object padlock = new object();
        public bool MultipleBrowserInstances { get; set; }
        public string DriverType { get; set; }
        public TestAdapter TestIdentifier { get; set; }
        public DateTime TestIdentifierStartTime { get; set; }
        public string Url { get; set; }
        public string ElementTimeout { get; set; }
        public string PageLoadTimeout { get; set; }
        public string EmailServiceUrl { get; set; }
        public string EmailServiceName { get; set; }
        public string EmailAddress { get; set; }
        public string TemporaryPassword { get; set; }
        public string GeneratedEmailAddress { get; set; }
        public string VerificationSenderAddress { get; set; }
        public string VerificationCode { get; set; }
        public string EmailSender { get; set; }
        public string ExpectedNumberOfEmails { get; set; }
        public string SetEmailFor { get; set; }
        public string EmailText { get; set; }
        public string FileNameSearchPatternEmail { get; set; }
        public string ExpectedNumberOfRepetition { get; set; }
        public string VerificationCodeText { get; set; }
        public List<EmailServiceParameters> EmailServices = new List<EmailServiceParameters>()
        {
            new EmailServiceParameters() { Name = "Mailinator", Url = "https://www.mailinator.com/" },
            new EmailServiceParameters() { Name = "Sharklasers", Url = "https://www.sharklasers.com/" }
        };

        public string SmsVerificationCode { get; set; }
        public static TestInMemoryParameters Instance
        {
            get
            {
                lock (padlock)
                {
                    if (instance == null)
                    {
                        instance = new TestInMemoryParameters();
                    }
                    return instance;
                }
            }
        }

        public string ApiEndpoint { get; set; }
        public JObject JsonApiResponse { get; set; }
        public XmlDocument XmlApiResponse { get; set; }
        public string TextApiResponse { get; set; }
        public string TestResultsDirectory { get; set; }
        #endregion
        #region Methods
        TestInMemoryParameters()
        {
            MultipleBrowserInstances = false;
            DriverType = EnvironmentVariable.GetEnvironmentVariable("browser", "Edge");
        }
        #endregion
    }
}

File: Automation/Woolsocks/TestAutomationFramework/src/main/java/common/core/TestInMemoryParameters.java
package common.core;

import common.core.configuration.TestConfiguration;
import org.openqa.selenium.remote.RemoteWebDriver;

import java.util.ArrayList;
import java.util.List;

public class TestInMemoryParameters {
    // region Fields and Properties
    private static TestInMemoryParameters instance = null;
    public boolean multipleBrowserInstances;
    public String elementTimeout;
    public String pageLoadTimeout;
    public RemoteWebDriver driver;
    public String emailAddress;
    public String emailServiceName;
    public String emailServiceUrl;
    public List<EmailServiceParameters> emailServices = new ArrayList<EmailServiceParameters>();
    public String generatedEmailAddress;
    public String emailSender;
    public String emailText;
    public String verificationSenderAddress;
    public String verificationCode;
    public String url;

    private String testResultsDirectory = "";
    private String apiEndpoint;
    private TestConfiguration testConfiguration;
    private String testRunParametersLocation;
    private String rootTestDirectory;

    // endregion
    // region Methods
    // Get thread safe instance
    public static TestInMemoryParameters getInstance() {
        if (instance == null) {
            synchronized (TestInMemoryParameters.class) {
                if (instance == null) {
                    instance = new TestInMemoryParameters();
                }
            }
        }
        return instance;
    }

    TestInMemoryParameters() {
        this.multipleBrowserInstances = false;
        this.elementTimeout = "60";
        this.pageLoadTimeout = "60";
    }

    public String getTestResultsDirectory() {
        return this.testResultsDirectory;
    }

    public void setTestResultsDirectory(String testResultsDirectory) {
        this.testResultsDirectory = testResultsDirectory;
    }

    public String getApiEndpoint() {
        return apiEndpoint;
    }

    public void setApiEndpoint(String apiEndpoint) {
        this.apiEndpoint = apiEndpoint;
    }

    public TestConfiguration getTestConfiguration() {
        return testConfiguration;
    }

    public void setTestConfiguration(TestConfiguration testConfiguration) {
        this.testConfiguration = testConfiguration;
    }

    public String getTestRunParametersLocation() {
        return testRunParametersLocation;
    }

    public void setTestRunParametersLocation(String testRunParametersLocation) {
        this.testRunParametersLocation = testRunParametersLocation;
    }

    public String getRootTestDirectory() {
        return rootTestDirectory;
    }

    public void setRootTestDirectory(String rootTestDirectory) {
        this.rootTestDirectory = rootTestDirectory;
    }

    // endregion
}
